TransWithoutContext.d.ts

```diff
@@ -7,14 +7,15 @@ type TransChild = React.ReactNode | Record<string, unknown>;
 export type TransProps<
   Key extends ParseKeys<Ns, TOpt, KPrefix>,
   Ns extends Namespace = _DefaultNamespace,
-  TOpt extends TOptions = {},
   KPrefix = undefined,
+  TContext extends string | undefined = undefined,
+  TOpt extends TOptions & { context?: TContext } = { context: TContext },
```

Comment:
I know this was closed, but if we remove the `{ context: TContext }` it works a pretty better,  like:
`TOpt extends TOptions & { context?: TContext }` in this way the IDE TS autocomplete works:
without context assignation: 
<img width="494" alt="image" src="https://github.com/i18next/react-i18next/assets/6584680/a4063ed7-82b5-4b32-8b66-956d0ba621be">
with context assignation:
<img width="493" alt="image" src="https://github.com/i18next/react-i18next/assets/6584680/d0916560-e4f1-4b4d-80e6-1475b7998332">

Reply:
Does it still correctly handle the typings for the context `property` of the `Trans` component?

Reply:
not exactly what I meant :D I meant more something like this:

<img width="669" alt="image" src="https://github.com/i18next/react-i18next/assets/3532342/4371d141-8c1d-4adf-8b28-659cd26f5179">
<img width="669" alt="image" src="https://github.com/i18next/react-i18next/assets/3532342/8abc16ed-1556-4392-abdf-a34c4c35bf88">